src/trace/step-function-service.ts

```diff
@@ -123,51 +127,65 @@ export class StepFunctionContextService {
   public get spanContext(): SpanContextWrapper | null {
     if (this.context === undefined) return null;
 
-    const traceId = this.deterministicMd5HashToBigIntString(this.context["step_function.execution_id"]);
-    const parentId = this.deterministicMd5HashToBigIntString(
+    const traceId = this.deterministicSha256HashToBigIntString(this.context["step_function.execution_id"], TRACE_ID);
+    const parentId = this.deterministicSha256HashToBigIntString(
       this.context["step_function.execution_id"] +
         "#" +
         this.context["step_function.state_name"] +
         "#" +
         this.context["step_function.state_entered_time"],
+      PARENT_ID,
     );
     const sampleMode = SampleMode.AUTO_KEEP;
+    const _DatadogSpanContext = require("dd-trace/packages/dd-trace/src/opentracing/span_context");
+    const id = require("dd-trace/packages/dd-trace/src/id");
 
-    const spanContext = SpanContextWrapper.fromTraceContext({
-      traceId,
-      parentId,
-      sampleMode,
-      source: TraceSource.Event,
+    const ddTraceContext = new _DatadogSpanContext({
+      traceId: id(traceId, 10),
+      spanId: id(parentId, 10),
+      sampling: { priority: sampleMode.toString(2) },
     });
 
+    const ptid = this.deterministicSha256HashToBigIntString(this.context["step_function.execution_id"], DD_P_TID);
+    ddTraceContext._trace.tags["_dd.p.tid"] = id(ptid, 10).toString(16);
+    const spanContext = new SpanContextWrapper(ddTraceContext, TraceSource.Event);
+
```

Comment:
Why are we not using the current `SpanContextWrapper` code? Is it because we need to add the ptid?

If so, is there another way to not require `dd-trace` code here? Or at least, we need to lazy load it, since the tracer could be not available

Reply:
1. Yes.
2. Lemme try to lazy load it.